base/rational.jl

```diff
@@ -264,7 +264,8 @@ for rel in (:<,:<=,:cmp)
                 end
                 return ($rel)(xc,yc)
             else
-                return xc > 0 ? ($rel)(xb,yb) : ($rel)(yb,xb)
+                xf, yf = xc > 0 ? (xb,yb) : (yb,xb)
+                return ($rel)(xf, yf)::Bool
```

Comment:
I don't know whether to do this or to use `@eval function ($rel)(x::$Tx, y::$Ty)::Bool` instead (which is available now). I thought it can be back-ported like this, but having 0.5 so close can make this possibility moot.